decompiler/frontend/binaryninja/frontend.py

```diff
@@ -123,30 +53,28 @@ def from_raw(cls, view: BinaryView):
         """Create a binaryninja frontend instance based on an initialized binary view."""
         return cls(view)
 
-    def create_task(self, function_identifier: Union[str, Function], options: Options) -> DecompilerTask:
-        """Create a task from the given function identifier."""
-        function = FunctionObject.get(self._bv, function_identifier)
-        tagging = CompilerIdiomsTagging(self._bv, function.function.start, options)
-        tagging.run()
+    def lift(self, task: DecompilerTask):
+        if task.failed:
```

Comment:
How can the task already be failed, if we did not lift it so far?

Reply:
It can't. When writing this code I didn't really think about this and more about the general contract of the lift method. From the typing itself, `lift` accepts any `DecompilerTask`. This means theoretically it can accept a failed task, which not lifting should be done for.
Assuming we don't move the creation of the task object back into the lift method, I think, even though practically this doesn't currently happen, it still makes sense to keep this future modifications could potentially make this possible.